model/task/db.go

```diff
@@ -616,6 +616,74 @@ func GetRecentTasks(period time.Duration) ([]Task, error) {
 	return tasks, nil
 }
 
+type StatsList struct {
+	Status string   `json:"status"`
+	Stats  []Result `json:"stats"`
+}
+
+func GetRecentTaskStatsList(period time.Duration, statKey string) ([]StatsList, error) {
```

Comment:
I don't completely follow this aggregation. Is the goal to get stats by status? If so, what's the reason for the second $group stage? This might be easier to walk through in person.